operator/pkg/helmreconciler/prune.go

```diff
@@ -211,8 +211,18 @@ func (h *HelmReconciler) GetPrunedResources(revision string, includeClusterResou
 			err = h.client.List(context.TODO(), objects,
 				client.MatchingLabelsSelector{Selector: s.Add(*componentRequirement)})
 		} else {
-			err = h.client.List(context.TODO(), objects,
-				client.MatchingLabelsSelector{Selector: selector.Add(*componentRequirement)})
+			// do not prune base components or addon components
+			excludeCN := []string{string(name.IstioBaseComponentName), string(name.AddonComponentName)}
```

Comment:
i'm concerned about not pruning everything in base. something may end up in base in the future that we _do_ want to prune so this selection seems too coarse. it would be better to be explicit about the kinds we don't want to prune. going through that exercise will help others comment on whether it's appropriate to prune/not prune a particular kind.

Reply:
The change is for uninstall command only actually and this path is executed by the uninstall, and I did not touch the shared logic used by the reconciler, so should be safe.